frontend/src/pages/CompareV2.tsx

```diff
@@ -51,22 +69,101 @@ function CompareV2(props: PageProps) {
   const runIds = (queryParamRunIds && queryParamRunIds.split(',')) || [];
 
   // Retrieves run details.
-  const { isError, data, refetch } = useQuery<ApiRunDetail[], Error>(
+  const {
+    isLoading: isLoadingRunDetails,
+    isError: isErrorRunDetails,
+    error: errorRunDetails,
+    data: runs,
+    refetch,
+  } = useQuery<ApiRunDetail[], Error>(
     ['run_details', { ids: runIds }],
     () => Promise.all(runIds.map(async id => await Apis.runServiceApi.getRun(id))),
     {
       staleTime: Infinity,
-      onError: async error => {
-        const errorMessage = await errorToMessage(error);
+    },
+  );
+
+  // Retrieves MLMD states (executions and linked artifacts) from the MLMD store.
+  const {
+    data: runArtifacts,
+    isLoading: isLoadingRunArtifacts,
+    isError: isErrorRunArtifacts,
+    error: errorRunArtifacts,
+  } = useQuery<RunArtifacts[], Error>(
+    ['run_artifacts', { runIds, runs }],
+    () => {
+      if (runs) {
+        return Promise.all(
+          runIds.map(async (r, index) => {
+            const context = await getKfpV2RunContext(r);
+            const executions = await getExecutionsFromContext(context);
+            const artifacts = await getArtifactsFromContext(context);
+            const events = (await getEventsByExecutions(executions)).filter(
+              e => e.getType() === Event.Type.OUTPUT,
+            );
+
+            // Match artifacts to executions.
+            const artifactMap = new Map();
+            artifacts.forEach(artifact => artifactMap.set(artifact.getId(), artifact));
+            const executionArtifacts = executions.map(execution => {
+              const executionEvents = events.filter(e => e.getExecutionId() === execution.getId());
+              const linkedArtifacts = executionEvents.map(event => {
+                return {
+                  event,
+                  artifact: artifactMap.get(event.getArtifactId()),
+                } as LinkedArtifact;
+              });
+              return {
+                execution,
+                linkedArtifacts,
+              } as ExecutionArtifacts;
+            });
+            return {
+              run: runs[index],
+              executionArtifacts,
+            };
+          }),
+        );
+      }
+      return [];
+    },
+    {
+      staleTime: Infinity,
+    },
+  );
+
+  useEffect(() => {
```

Comment:
I updated the PR here (https://github.com/kubeflow/pipelines/pull/7897/commits/63bad148129f6c9d0c2ad752611b38935404351d) to centralize the banner display process, similar to the idea here: https://github.com/kubeflow/pipelines/blob/146dae79cfa6ba2b45c524a7ca58dd610ea6ed8c/frontend/src/pages/Compare.tsx#L77-L116

Since I will add another `useQuery` in the next PR when calling `getArtifactTypes`, I wanted to have all of the banner updates in one place instead of strewn across different `useQuery` calls where those banner updates could overlap. Let me know if this is a good idea or if I should revert some of these changes (since it may increase code complexity).